sematic/api_client.py

```diff
@@ -72,13 +70,52 @@ def get_artifact_value_by_id(
     Any
         The value of the requiested artifact.
     """
-    # TODO: Store storage type on artifact
-    if storage is None:
-        storage = S3Storage()
-
     artifact = _get_artifact(artifact_id)
 
-    return get_artifact_value(artifact, storage)
+    return get_artifact_value(artifact)
+
+
+def get_artifact_value(artifact: Artifact) -> Any:
+    storage_engine, location = _get_artifact_location(artifact.id, mode=StorageMode.READ)
+    payload = storage_engine.get(location)
+
+    return deserialize_artifact_value(artifact, payload)
+
+
+def store_artifact_value(artifact_id: str, value: bytes):
+    storage_engine, location = _get_artifact_location(artifact_id, StorageMode.WRITE)
+    storage_engine.set(location, value)
+
+
+def _get_artifact_location(artifact_id: str, mode: StorageMode) -> Tuple[Storage, str]:
+    response = _get(f"/artifacts/{artifact_id}/location/{mode.value}")
+
+    return _get_location_for_response(response)
+
+
+def store_future_pickle(future_id: str, pickle: bytes):
```

Comment:
This should be agnostic of the serialization method, as it get already-serialized bytes, so it shouldn't include "pickle" in the name.
Also, it should be descriptive of the contents.
```suggestion
def store_future_artifact_bytes(future_id: str, pickle: bytes):
```

Reply:
Done.